src/mixins/dependent.js

```diff
@@ -0,0 +1,60 @@
+export function factory (opts = { closeDependents: true }) {
+  return {
+    props: {
+      closeDependents: {
+        type: Boolean,
+        default: opts.closeDependents
+      },
+      isDependent: {
+        type: Boolean,
+        default: opts.isDependent
+      }
+    },
+
+    methods: {
+      getOpenDependents () {
+        const results = []
+        if (this.closeDependents) {
+          const search = (children) => {
+            for (const child of children) {
+              if (child.isActive && (child.isDependent || (child.closeDependents && child.isDependent !== false))) {
+                results.push(child)
+              } else {
+                search(child.$children)
+              }
+            }
+          }
+          search(this.$children)
+        }
+        return results
+      },
+      getOpenDependentElements () {
+        const result = []
+        for (const dependent of this.getOpenDependents()) {
+          result.push(...(dependent.getClicableDependentElements()))
+        }
+        return result
```

Comment:
could this not be:

`
return this.getOpenDependents().map(d => d.getClickableDependentElements())
`